src/webview/components/kdbDataSourceView.ts

```diff
@@ -32,24 +32,83 @@ import {
   filterOperators,
 } from "../../models/dataSource";
 import { MetaObjectPayload } from "../../models/meta";
-import { kdbStyles, vscodeStyles } from "./styles";
 import { DataSourceCommand, DataSourceMessage2 } from "../../models/messages";
+import { shoelaceStyles } from "./styles";
 
 const MAX_RULES = 32;
 
 @customElement("kdb-data-source-view")
 export class KdbDataSourceView extends LitElement {
   static styles = [
-    vscodeStyles,
-    kdbStyles,
+    shoelaceStyles,
     css`
-      vscode-text-area::part(control) {
-        width: 48em !important;
+      .container {
+        display: flex;
+        flex-flow: row nowrap;
+        overflow-x: auto;
+        gap: var(--sl-spacing-medium);
+        padding-left: var(--sl-spacing-medium);
+        padding-right: var(--sl-spacing-medium);
       }
 
-      .panel {
-        width: 50em;
-        margin-top: 0.5em;
+      .tabs {
+        flex-grow: 1;
+      }
+
+      .actions {
+        display: flex;
+        flex-flow: column nowrap;
+        flex-grow: 0;
+        gap: var(--sl-spacing-x-small);
+        margin-top: calc(
+          1rem * var(--sl-line-height-dense) + 3 * var(--sl-spacing-medium)
+        );
+      }
+
+      .actions sl-button-group > sl-button {
+        flex-grow: 1;
+      }
+
+      sl-tab::part(base) {
+        padding: var(--sl-spacing-medium);
+      }
+
+      sl-select::part(listbox) {
+        min-width: max-content;
+      }
+
+      sl-tab-panel {
+        --padding: 0;
+        overflow-y: scroll;
+        padding-top: var(--sl-spacing-medium);
+        padding-bottom: var(--sl-spacing-medium);
+        height: calc(
+          100vh - 1rem * var(--sl-line-height-dense) - 2 *
+            var(--sl-spacing-medium)
+        );
+      }
+
+      sl-input,
+      sl-select {
+        min-width: 13rem;
+        max-width: 13rem;
+      }
+
+      sl-checkbox {
+        padding-bottom: var(--sl-spacing-2x-small);
+      }
+
+      .col {
+        display: flex;
+        flex-direction: column;
+        gap: var(--sl-spacing-x-small);
+      }
+
+      .row {
+        display: flex;
+        flex-direction: row;
+        gap: var(--sl-spacing-x-small);
+        align-items: flex-end;
```

Comment:
should be a good idea to move it to styles too, add as DSStyles or something

Reply:
Moved the styles